File: SDKs/twilio.com/twilio_oauth_v1/1.37.3/python/src/sdk/models/shared/oauth_v1_openid_discovery.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class OauthV1OpenidDiscovery:
    authorization_endpoint: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('authorization_endpoint') }})
    claims_supported: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('claims_supported') }})
    device_authorization_endpoint: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('device_authorization_endpoint') }})
    id_token_signing_alg_values_supported: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id_token_signing_alg_values_supported') }})
    issuer: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('issuer') }})
    jwk_uri: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('jwk_uri') }})
    response_type_supported: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('response_type_supported') }})
    revocation_endpoint: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('revocation_endpoint') }})
    scopes_supported: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('scopes_supported') }})
    subject_type_supported: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subject_type_supported') }})
    token_endpoint: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('token_endpoint') }})
    url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('url') }})
    userinfo_endpoint: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('userinfo_endpoint') }})
    

File: SDKs/twilio.com/twilio_oauth_v1/1.37.3/typescript/src/sdk/models/shared/oauthv1openiddiscovery.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class OauthV1OpenidDiscovery extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=authorization_endpoint" })
  authorizationEndpoint?: string;

  @SpeakeasyMetadata({ data: "json, name=claims_supported" })
  claimsSupported?: string[];

  @SpeakeasyMetadata({ data: "json, name=device_authorization_endpoint" })
  deviceAuthorizationEndpoint?: string;

  @SpeakeasyMetadata({ data: "json, name=id_token_signing_alg_values_supported" })
  idTokenSigningAlgValuesSupported?: string[];

  @SpeakeasyMetadata({ data: "json, name=issuer" })
  issuer?: string;

  @SpeakeasyMetadata({ data: "json, name=jwk_uri" })
  jwkUri?: string;

  @SpeakeasyMetadata({ data: "json, name=response_type_supported" })
  responseTypeSupported?: string[];

  @SpeakeasyMetadata({ data: "json, name=revocation_endpoint" })
  revocationEndpoint?: string;

  @SpeakeasyMetadata({ data: "json, name=scopes_supported" })
  scopesSupported?: string[];

  @SpeakeasyMetadata({ data: "json, name=subject_type_supported" })
  subjectTypeSupported?: string[];

  @SpeakeasyMetadata({ data: "json, name=token_endpoint" })
  tokenEndpoint?: string;

  @SpeakeasyMetadata({ data: "json, name=url" })
  url?: string;

  @SpeakeasyMetadata({ data: "json, name=userinfo_endpoint" })
  userinfoEndpoint?: string;
}
